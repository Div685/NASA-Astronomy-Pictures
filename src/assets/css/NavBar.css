.nav {
  position: fixed;
  top: 0;
  width: 100%;
  padding: 30px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 30px;
  z-index: 1;

  /* Animation */
  transition: all 0.5s;
  transition-timing-function: ease-in;
}

.nav__black {
  background-color: #1a1a29;
  padding: 30px;
}

.nav__logo {
  position: fixed;
  left: 20px;
  width: 80px;
  color: #fff;
  top: 8px;
}

.nav__link {
  position: fixed;
  right: 20px;
  color: #fff;
}
