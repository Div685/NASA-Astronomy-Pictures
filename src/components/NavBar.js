import React, { useState, useEffect } from 'react';
import '../assets/css/NavBar.css';

function NavBar() {
  const [show, handleShow] = useState(false);

  useEffect(() => {
    window.addEventListener('scroll', () => {
      if (window.scrollY > 100) {
        handleShow(true);
      } else {
        handleShow(false);
      }
    });
    return () => {
      window.removeEventListener('scroll', null);
    };
  }, []);

  return (
    <div className={`nav ${show && 'nav__black'}`}>
      <a href="/" className="nav__logo">
        <h3 className="nav__logo">AstroSplash</h3>
      </a>

      <span className="nav__link">NASA APOD API</span>
    </div>
  );
}

export default NavBar;
