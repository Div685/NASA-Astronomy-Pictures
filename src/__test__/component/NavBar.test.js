import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import NavBar from '../../components/NavBar';
import '@testing-library/jest-dom';

describe('NavBar Component', () => {
  it('renders NavBar page correctly', () => {
    const tree = renderer.create(<NavBar />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Test for NavBar component', () => {
  it('test the heading of nav component', () => {
    render(<NavBar />);
    expect(screen.getByRole('heading')).toHaveTextContent('AstroSplash');
  });
});
