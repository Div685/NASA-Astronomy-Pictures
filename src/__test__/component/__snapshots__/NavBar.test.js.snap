// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders NavBar page correctly 1`] = `
<div
  className="nav false"
>
  <a
    className="nav__logo"
    href="/"
  >
    <h3
      className="nav__logo"
    >
      AstroSplash
    </h3>
  </a>
  <span
    className="nav__link"
  >
    NASA APOD API
  </span>
</div>
`;
